### ISSUE TITLE
PdoSessionHandler with Oracle not working as expected

### ISSUE THREAD

**Issue by elislenio** [symfony/symfony#18305]
When using PdoSessionHandler with Oracle, Symfony 2.8 raises an exception:

> Warning: PDOStatement::execute(): supplied argument is not a valid stream resource
> 500 Internal Server Error - ContextErrorException

Stack trace starts with:
`in vendor\symfony\symfony\src\Symfony\Component\HttpFoundation\Session\Storage\Handler\PdoSessionHandler.php at line 538`

When switching to mysql all works fine.

**Comment by Tobion** [comment]
It seems to be the handling of `PDO::PARAM_LOB` in https://github.com/symfony/symfony/blob/master/src/Symfony/Component/HttpFoundation/Session/Storage/Handler/PdoSessionHandler.php#L535
Oracle might expect a stream instead of a string. But others work with string. So it's not clear how to make it cross-dbms functional.

What PHP and oracle client and server version do you have?

**Comment by elislenio** [comment]
I have PHP 5.5.12 and Oracle XE 11.2.0 (client 11.2 too)

Looking arround this issue:
Creating the stream does the trick and dismisses the warning:
`$stream = fopen('php://memory', 'r+');
fwrite($stream, $data);
rewind($stream);`
(the stream works well with mysql too)

once passed this step, it shows up a second issue: session data is empty on the database. I think it is the merge statement not handling well the BLOB field, since merge does not support a RETURNING clause.
Instead of merge I used insert and update statements:
`INSERT INTO $this->table ($this->idCol, $this->dataCol, $this->lifetimeCol, $this->timeCol) VALUES (:id, EMPTY_BLOB(), :lifetime, :time) RETURNING $this->dataCol into :data`
`UPDATE $this->table SET $this->dataCol = EMPTY_BLOB(), $this->lifetimeCol = :lifetime, $this->timeCol = :time WHERE $this->idCol = :id RETURNING $this->dataCol into :data`

[http://www.oracle.com/technetwork/articles/fuecks-lobs-095315.html](url)

After trying these changes, the session data get inserted and update well in the database.

Finally, when I thought that everything was going smoothly, I found another problem related to the loss of the authentication token. Perhaps is related to this issue:
[https://github.com/symfony/symfony/issues/14569](url)

It seems that the implementation of PdoSessionHandler for Oracle requires further review.
